### PR TITLE
Catch FPEs on stm32, raise InternalError on same

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by tridge** [comment]
nice1 I think we need a test to run on stm32 boards (F4, F7, H7) to check the behavior. We need to make sure we keep flying after a FPE and get the right Inf etc

**Comment by peterbarker** [comment]
@tridge I used a patch on a bench-test Pixhawk which did a floating-point division-by-zero when switching into loiter.  The Pixhawk continued to run.

I was considering adding an RC option to do it - but that's starting to feeling the "everything looks like a nail" part of ArduPilot.

I was wondering whether we should *not* clear these flags, just look at them.  That would be a little safer but lose the ability to count the number of floating point exceptions occurring.

**Comment by peterbarker** [comment]
.... I should also point out that I'm not attempting to catch/handle the exception.  These flags appear to be set regardless of any vectors, and clearing them shouldn't affect anything other than anybody else trying to monitor them.

**Comment by WickedShell** [comment]
Worth testing with some simple lua scripts that would provoke these. I could easily see them creating a FPE which we then would be panicing the user with, but since it's a script it isn't nearly as worrisome.

**Comment by peterbarker** [comment]
Stopped clearing the FPE registers as per devcall discussions.

File: libraries/AP_HAL_ChibiOS/HAL_ChibiOS_Class.cpp [hunk 1]
```diff
@@ -254,6 +255,12 @@ static void main_loop()
         }
 #endif
         schedulerInstance.watchdog_pat();
+#if !defined(HAL_BOOTLOADER_BUILD) && !defined(IOMCU_FW)
+        static uint32_t stm32_old_fpe_flags;
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/HAL_ChibiOS_Class.cpp]: don't need this variable now, can just check against zero

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/HAL_ChibiOS_Class.cpp]: Fixed.

**Pull request closed without merging**